### PR TITLE
KEYCLOAK-18268 logo_uri, policy_uri  and tos_uri ( client attributes)

### PR DESCRIPTION
**Pull request by cgeorgilakis**
Keycloak Documentation PR : https://github.com/keycloak/keycloak-documentation/pull/1181.
Needed for this [PR](https://github.com/keycloak/keycloak/pull/8092).

### CODE REVIEW

**Comment by cgeorgilakis** [comment]
Our goal was to do no database changes. However, for logo_uri some SAML clients use data:image and the logo_uri length is over 4000 characters. Current database length for VALUE field of CLIENT_ATTRIBUTES is insufficient.
We believe that we should create a new field in ClientEntity for logo_uri with text format. Do you agree or you would like to propose a different solution?

File: services/src/main/java/org/keycloak/validation/DefaultClientValidationProvider.java [hunk 1]
```diff
@@ -180,6 +199,24 @@ private void checkUri(FieldMessages field, String url, ValidationContext<ClientM
...
 
+    private void checkUriLogo(FieldMessages field, String url, ValidationContext<ClientModel> context) {
+        if (url == null || url.isEmpty()) {
+            return;
+        }
+
+        try {
+            URI uri = new URI(url);
+
+            if (uri.getScheme() != null &&  uri.getScheme().equals("javascript")) {
```
Comment (jbman) [inline services/src/main/java/org/keycloak/validation/DefaultClientValidationProvider.java:210]: This should be more strict: "Schemes other than “https”, “http”, or “data” SHOULD NOT be used."
according to https://docs.oasis-open.org/security/saml/Post2.0/sstc-saml-metadata-ui/v1.0/os/sstc-saml-metadata-ui-v1.0-os.html#__RefHeading__10407_1021935550

Comment (cgeorgilakis) [reply services/src/main/java/org/keycloak/validation/DefaultClientValidationProvider.java:210]: It is not mandatory. Moreover, as I know OIDC specification has nothing similar. I want validator to be common for OIDC and SAML ( as in other fields) and to be similar to other fields validator - only logo_uri can have data scheme.
However, if Keycloak team want, I can add proposed validator only for SAML client.

**Comment by guymoyo** [comment]
> CLIENT_ATTRIBUTES

Recently there were some requirements where REALM_ATTRIBUTE needed to be extended, the same can be applied in CLIENT_ATTRIBUTE.

`<dropColumn tableName="REALM_ATTRIBUTE" columnName="VALUE"/>
        <renameColumn tableName="REALM_ATTRIBUTE" oldColumnName="VALUE_NEW" newColumnName="VALUE" columnDataType="NCLOB"/>`

**Pull request closed without merging**